components/datetime/src/format/time_zone.rs

```diff
@@ -41,271 +67,53 @@ where
     T: TimeZoneInput,
 {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
```

Comment:
Nit: This function should just call `self.write_to(f)`.

If you do this, then do we even still need `TimeZoneFormatKind` any more? It would be really great if we could get rid of that. I think also that this is the only call site for `write_pattern`. See if you can delete some code! :)

Reply:
good catch. It appears that we can remove `write_pattern` and `TimeZoneFormatKind`.

But I prefer not to make any new big changes in this PR, and remove `write_pattern` and `TimeZoneFormatKind` in next PR.